intake_xarray/base.py

```diff
@@ -52,17 +63,16 @@ def read_partition(self, i):
         if not isinstance(i, (tuple, list)):
             raise TypeError('For Xarray sources, must specify partition as '
                             'tuple')
-        if isinstance(i, list):
-            i = tuple(i)
-        if hasattr(self._ds, 'variables') or i[0] in self._ds.coords:
-            arr = self._ds[i[0]].data
-            i = i[1:]
+        variable, *part = i
+        part = tuple(part)
+        if hasattr(self._ds, 'variables') or variable in self._ds.coords:
+            arr = self._ds[variable].data
         else:
             arr = self._ds.data
         if isinstance(arr, np.ndarray):
-            return arr
-        # dask array
-        return arr.blocks[i].compute()
+            # Make a dask.array so that we can return the appropriate block.
+            arr = dask.array.from_array(arr, chunks=self._chunks[variable])
```

Comment:
Even if we have a dask array here, can we be sure that it has the same chunks as the one that the client will have? It looks like encoding a numpy-backed `xarray.Dataset` with `to_zarr(...)` prompts zarr to automatically choose a chunking for us. If a serialized a _dask_ -backed `xarray.Dataset`, will `to_zarr(...)` respect the existing chunking? If yes, then we have no problem here.

Reply:
`to_zarr` uses the current chunking; rechunking is optional.